collector/load_balancer_collector_test.go

```diff
@@ -0,0 +1,234 @@
+						{
+							ID:     "fake-load-balancer-pool-id-02",
+							Status: 0.0,
+							MembersStatus: []loadBalancerPoolMemberStatusMetric{
+								{
+									IPAddress: fakeLoadbalancerPoolMemberIP,
+									Port:      fakeLoadbalancerPoolMemberPort,
+									Status:    0.0,
+								},
+							},
+						},
+					},
+				}, {
+					ID:     "fake-load-balancer-id-03",
+					Name:   "fake-load-balancer-name-03",
+					Status: 0.0,
+					PoolsStatus: []loadBalancerPoolStatusMetric{
+						{
+							ID:     "fake-load-balancer-pool-id-03",
+							Status: 0.0,
+							MembersStatus: []loadBalancerPoolMemberStatusMetric{
+								{
+									IPAddress: fakeLoadbalancerPoolMemberIP,
+									Port:      fakeLoadbalancerPoolMemberPort,
+									Status:    0.0,
+								},
+							},
+						},
+					},
+				}, {
+					ID:     "fake-load-balancer-id-04",
+					Name:   "fake-load-balancer-name-04",
+					Status: 0.0,
+					PoolsStatus: []loadBalancerPoolStatusMetric{
+						{
+							ID:     "fake-load-balancer-pool-id-04",
+							Status: 0.0,
+							MembersStatus: []loadBalancerPoolMemberStatusMetric{
+								{
+									IPAddress: fakeLoadbalancerPoolMemberIP,
+									Port:      fakeLoadbalancerPoolMemberPort,
+									Status:    0.0,
+								},
+							},
+						},
+					},
+				}, {
+					ID:     "fake-load-balancer-id-05",
+					Name:   "fake-load-balancer-name-05",
+					Status: 0.0,
+					PoolsStatus: []loadBalancerPoolStatusMetric{
+						{
+							ID:     "fake-load-balancer-pool-id-05",
+							Status: 0.0,
+							MembersStatus: []loadBalancerPoolMemberStatusMetric{
+								{
+									IPAddress: fakeLoadbalancerPoolMemberIP,
+									Port:      fakeLoadbalancerPoolMemberPort,
+									Status:    0.0,
+								},
+							},
+						},
+					},
+				}, {
+					ID:     "fake-load-balancer-id-06",
+					Name:   "fake-load-balancer-name-06",
+					Status: 0.0,
+					PoolsStatus: []loadBalancerPoolStatusMetric{
+						{
+							ID:     "fake-load-balancer-pool-id-06",
+							Status: 0.0,
+							MembersStatus: []loadBalancerPoolMemberStatusMetric{
+								{
+									IPAddress: fakeLoadbalancerPoolMemberIP,
+									Port:      fakeLoadbalancerPoolMemberPort,
+									Status:    1.0,
+								},
+							},
+						},
+					},
+				}, {
+					ID:     "fake-load-balancer-id-07",
+					Name:   "fake-load-balancer-name-07",
+					Status: 0.0,
+					PoolsStatus: []loadBalancerPoolStatusMetric{
+						{
+							ID:     "fake-load-balancer-pool-id-07",
+							Status: 1.0,
+							MembersStatus: []loadBalancerPoolMemberStatusMetric{
+								{
+									IPAddress: fakeLoadbalancerPoolMemberIP,
+									Port:      fakeLoadbalancerPoolMemberPort,
+									Status:    0.0,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testcases {
+		mockLoadBalancerClient := &mockLoadBalancerClient{
+			responses: tc.loadBalancerResponses,
+		}
+		var loadBalancers []loadbalancer.LbService
```

Comment:
nit: line 221-228 can be extracted as function since this is technically mock response generator, can be reused by other tests if any

Reply:
ack, extracted into method